Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -948,7 +948,7 @@ void LocalDOMWindow::processPostMessage(JSC::JSGlobalObject& lexicalGlobalObject
         if (!globalObject)
             return;
 
-        auto& vm = globalObject->vm();
+        Ref vm = globalObject->vm();
```

Comment:
cc @rniwa . I think the JSC team has strong feelings about not ref'ing the VM because it's unnecessary. What's our plan?

Reply:
If we rename WebCore::commonVM() to WebCore::vmSingleton(), then static analysis will recognize that it doesn't need refcounting.

Then, we can change JSDOMWindowBase to:
* initialize itself using WebCore::commonVM(), instead of taking an arbitrary VM& argument (or, maybe as an initial stopgap, just RELEASE_ASSERT that they're the same);
* offer a vmSingleton() member function that returns m_vm (and maybe debug ASSERTS that they're the same);

Then, in places like this, as long as we use the right type, we can skip refcounting. And if we don't have a way to use the right type, then we don't have a compile-time proof that we can skip refcounting, so we should do the refcounting.

Reply:
Since this is just one rare refcount, maybe we can let it slide for now, though.

Reply:
We should probably revert this change. I think it's better to fix these in en masse.

Reply:
Indeed there are a lot of warnings about auto& vm = globalObject->vm();, a very frequent construction (even besides the generated code). Will revert that part of the PR.

Reply:
Since the vm changes were more frequent than I remembered (I thought it was just one), I removed them all and added some new fixes for clang-tidy warnings.